### PR TITLE
Integration-level configuration

### PR DESCRIPTION
**Pull request by labbati**
### Description

In order to enabled Trace Analytics configuration from clients, we need to provide a way to configure integrations individually instead of globally.

This PR is an attempt at proposing a way to do it in a quite structured and flexible way.

### Readiness checklist
- [x] [Changelog entry](docs/changelog.md) added, if necessary
- [x] Tests added for this feature/bug

### CODE REVIEW

**Comment by labbati** [comment]
While tests are broken (also on master) this can be reviewed in the meantime while we wait for merge until master is ok.

**Comment by labbati** [comment]
@pawelchcki I applied what we discussed and (at least to me 😄) it makes much more sense than my original implementation. Whenever you have a minute it would be great to have a second round of CR

File: src/DDTrace/Integrations/AbstractIntegrationConfiguration.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    private $integrationName;
+
+    /**
+     * @param string $integrationName
+     */
+    public function __construct($integrationName)
+    {
+        $this->integrationName = $integrationName;
+        $prefix = strtoupper(str_replace('-', '_', trim($this->integrationName)));
+        $this->registry = new EnvVariableRegistry("DD_${prefix}_");
```
Comment (pawelchcki) [inline src/DDTrace/Integrations/AbstractIntegrationConfiguration.php]: ❤️

File: tests/Unit/Integrations/AbstractIntegrationConfigurationTest.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        $config = new DummyIntegrationConfiguration('dummy');
+        $this->assertSame(0.8, $config->getSampleFloat());
+    }
+
+    public function testFloatValueDefault()
+    {
+        $config = new DummyIntegrationConfiguration('dummy');
+        $this->assertSame(1.23, $config->getSampleFloat());
+    }
+}
```
Comment (pawelchcki) [inline tests/Unit/Integrations/AbstractIntegrationConfigurationTest.php]: I :heart: the contents of this file

**Pull request merged**